pylasu/validation/validation.py

```diff
@@ -35,7 +35,11 @@ def semantic(message: str, severity: IssueSeverity = IssueSeverity.ERROR, positi
         return Issue(IssueType.SEMANTIC, message, severity, position)
 
 
+class WithIssues:
+    """Many classes have the necessity of tracking issues"""
+    issues: List[Issue] = field(default_factory=list)
```

Comment:
To avoid having to repeat the property for each subclass, we could change this to `... = []` rather than using the `default_factory` which is used for dataclasses. Defining this same class as dataclass would instead induce errors with the constructors.

Reply:
I agree, then each subclass should be responsible of explicitly invoking the `WithIssues` constructor. If defining `WithIssues` as `@dataclass`, each field in the subclasses should be defined as optional and explicit names in creating new objects should be used to avoid confusion (`issues` would be the first constructor parameter).